File: medplum-java/medplum-model/src/main/java/com/fhir/Visionprescription_prismBase.java
package com.fhir;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Locale;
import javax.annotation.Nonnull;

public final class Visionprescription_prismBase {
  public static final Visionprescription_prismBase IN = new Visionprescription_prismBase(Value.IN, "IN");

  public static final Visionprescription_prismBase UP = new Visionprescription_prismBase(Value.UP, "UP");

  public static final Visionprescription_prismBase OUT = new Visionprescription_prismBase(Value.OUT, "OUT");

  public static final Visionprescription_prismBase DOWN = new Visionprescription_prismBase(Value.DOWN, "DOWN");

  private final Value value;

  private final String string;

  Visionprescription_prismBase(Value value, String string) {
    this.value = value;
    this.string = string;
  }

  public Value getEnumValue() {
    return value;
  }

  @Override
  @JsonValue
  public String toString() {
    return this.string;
  }

  @Override
  public boolean equals(Object other) {
    return (this == other) 
      || (other instanceof Visionprescription_prismBase && this.string.equals(((Visionprescription_prismBase) other).string));
  }

  @Override
  public int hashCode() {
    return this.string.hashCode();
  }

  public <T> T accept(Visitor<T> visitor) {
    switch (value) {
      case IN:
        return visitor.visitIN();
      case UP:
        return visitor.visitUP();
      case OUT:
        return visitor.visitOUT();
      case DOWN:
        return visitor.visitDOWN();
      case UNKNOWN:
      default:
        return visitor.visitUnknown(string);
    }
  }

  @JsonCreator(
      mode = JsonCreator.Mode.DELEGATING
  )
  public static Visionprescription_prismBase valueOf(@Nonnull String value) {
    String upperCasedValue = value.toUpperCase(Locale.ROOT);
    switch (upperCasedValue) {
      case "IN":
        return IN;
      case "UP":
        return UP;
      case "OUT":
        return OUT;
      case "DOWN":
        return DOWN;
      default:
        return new Visionprescription_prismBase(Value.UNKNOWN, upperCasedValue);
    }
  }

  public enum Value {
    UP,

    DOWN,

    IN,

    OUT,

    UNKNOWN
  }

  public interface Visitor<T> {
    T visitUP();

    T visitDOWN();

    T visitIN();

    T visitOUT();

    T visitUnknown(String unknownType);
  }
}

File: medplum-typescript/src/model/fhir/Visionprescription_prismBase.ts
export type Visionprescription_prismBase =
    | Visionprescription_prismBase.Up
    | Visionprescription_prismBase.Down
    | Visionprescription_prismBase.In
    | Visionprescription_prismBase.Out;

export const Visionprescription_prismBase = {
    Up: "up" as Visionprescription_prismBase.Up,
    Down: "down" as Visionprescription_prismBase.Down,
    In: "in" as Visionprescription_prismBase.In,
    Out: "out" as Visionprescription_prismBase.Out,

    _visit: <Result>(value: Visionprescription_prismBase, visitor: Visionprescription_prismBase._Visitor<Result>): Result => {
        switch (value) {
            case Visionprescription_prismBase.Up: return visitor.up();
            case Visionprescription_prismBase.Down: return visitor.down();
            case Visionprescription_prismBase.In: return visitor.in();
            case Visionprescription_prismBase.Out: return visitor.out();
            default: return visitor._unknown();
        }
    },
};

export declare namespace Visionprescription_prismBase {
    export type Up = "up" & {
        __Up: void,
    };
    export type Down = "down" & {
        __Down: void,
    };
    export type In = "in" & {
        __In: void,
    };
    export type Out = "out" & {
        __Out: void,
    };

    export interface _Visitor<Result> {
        up: () => Result;
        down: () => Result;
        in: () => Result;
        out: () => Result;
        _unknown: () => Result;
    }
}
